### ISSUE TITLE
Details of total batch size

### ISSUE THREAD

**Issue by luofuli** [LiyuanLucasLiu/Transformer-Clinic#6]
Hi, I find the following experimental setting details in your paper. And the max-token is 3584 in each gpu. How many GPUs you use to train En-De and En-Fr? I want to reproduce the results with the same total batch size over all gpus.

> For FR, #warmup steps is 8000, max #epochs is 50, and peak learning rate is 0.0007. For DE, #warmup steps is 4000, max #epochs is 50, and learning rate is 0.001. Max #tokens in each batch is set to 3584 following

**Comment by LiyuanLucasLiu** [comment]
Thanks for asking. The total batch size is the product of update frequency, max token # and gpu number. For example, we use four gpus on En-De, 8192 tokens (maximum) on each, and update at evey batch, which leads to 32768 tokens. The actual tokens per batch is about 30000 tokens. Training recipes are available at https://github.com/LiyuanLucasLiu/Transformer-Clinic/tree/master/nmt-experiments. 

BTW, 4096 is the maximum token number for IWSLT (we would update these numbers in the paper). In our experiments, the peak learning rate and other hyper-parameters are set following the previous work. I list some references below: 
* En-Fr: https://github.com/pytorch/fairseq/issues/233
* En-De: https://github.com/zhuohan123/macaron-net/blob/master/translation/macaron-scripts/train/train-wmt14-base.sh
